Rebaseable master

Nadahar opened the pull request:
This isn't a PR that's meant for merging, it's just here to have a place to easily see and discuss the changes.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/digitalmediaserver/digitalmediaserver/19)
<!-- Reviewable:end -->

Nadahar commented:
@Sami32 Have you had the time to look at the MediaInfo logging?

---

Review status: 0 of 10 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/digitalmediaserver/digitalmediaserver/19#-:-KvwbMzSHB0bjIzv7oYd:b9owbr0)*
<!-- Sent from Reviewable.io -->

Sami32 commented:
Yes, i briefly tested it, and it does look good. I will say 90% ;)
I will look at it tomorrow. Nothing hurry, isn't ?
:+1:

Nadahar commented:
@Sami32 No hurry, I just wanted to know if that was what you wanted. It's easy to adjust the columns, the number of colums, the labels etc - so you should be able to tweak it to something you like.

Sami32 commented:
I could try, i guess that after some tests and try i will finish by succeed, but all your alarms will waken your family for some hours or days :wink:

Nadahar commented:
@Sami32 After you gave me write access to your repos I get email's when you push there as well, so it won't do any difference ;)

Don't get me wrong, I can do the changes you have pointed out - it's just that I never meant for it to be exactly like that, it was more an example and a "framework" for building the log information.

Sami32 commented:
OK, i just tried to avoid make you bored by 20 commits spaced by 5mn, so i prefered keep "quiet", also the fact that my mind is busy by other HTTP stuff that make me crazy...
Anyway, i'll need to take time to look at which field have been omitted, before pushing anything...

Sami32 commented:
I didn't thought about that, so you are lucky that i don't work on this repository by now, or you'll get soon crazy LOL

Sami32 commented on CONTRIBUTING.md:
```diff
@@ -1,23 +1,26 @@
+* Click `Pack debug files` on the lower left.
+* Click `Zip selected files`.
+* Save the zip file to a location you will remember.
+* Attach the zip file in the corresponding GitHub issue.
```
It seem that we have forgotten the DMS headless users ;)

Nadahar replied:
@Sami32 True, although all we can tell them is to attach the log file and the config file. The problem is that neither of these are in "standard locations", so it can be difficult to give a universal explanation.

The profile folder can end up in many different locations depending on the OS/installation on Linux, where this is most relevant.

Sami32 replied:
Just naming the 2/3 files need should be suffisant, as i consider that every user should at least be able to know how to run a search on his OS.

Nadahar replied:
@Sami32 Please check the latest update.

Sami32 replied:
The latest...well i wan always try ;P

Nadahar commented:
Done

---

Review status: 0 of 12 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/digitalmediaserver/digitalmediaserver/19#-:-Kz-qXmEZ2dVpBooDyIG:bb6un1j)*
<!-- Sent from Reviewable.io -->

Sami32 commented on CONTRIBUTING.md:
```diff
@@ -1,23 +1,35 @@
+* Attach the zip file in the corresponding GitHub issue.
 
-Thanks for your efforts in helping us make DMS better!
+If it is a new bug that wasn't there in a previous version, we will often request debug information from the last version that did not have the bug too. If that is the case, please repeat the above steps for the last version without the bug as well.
```
A comma should be added as following:
`please, repeat`

Yeah, good point :+1:

Nadahar replied:
@Sami32 I disagree - if I move the comma to after please, the sentence will be very strange - or am I missing something again?

Sami32 replied:
Damn...
I meant:
`If it is a new bug that wasn't there in a previous version, we will often request debug information from the last version that did not have the bug too. If that is the case, please, repeat the above steps for the last version without the bug as well.`

Nadahar replied:
@Sami32 Still wrong, it must be French 🤖

Nadahar replied:
@Sami32 Actually, "please" can simply be removed.

Sami32 replied:
Ok. Yes, frenches do it.

I'm feeling that you remove things when i set a comment .It's a nice word, but i'm not a english native so...

Nadahar replied:
@Sami32 Ok, I just leave it as it is then.

Sami32 commented on CONTRIBUTING.md:
```diff
@@ -1,23 +1,35 @@
-For more information, please see STYLEGUIDE.md
+# <a name="Attachments"></a>Attachments
+
+GitHub only allows a few attachment extensions: `.png`, `.gif`, `.jpg`, `.docx`, `.pptx`, `.xlsx`, `.txt`, `.pdf`, `.zip` and `.gz`. Other attachments can be zipped and attached here. Text files can also simply have a `.txt` extension can be added to their file name.
```
>Text files can also simply have a `.txt` extension can be added to their file name.

Something wrong here. I guess that you meant to say "Files can also..."

Nadahar replied:
@Sami32 Something is wrong. I didn't mean to say "fake" though, as that can be confusing. A ```.txt``` extension for a text file is hardly fake :P

Sami32 replied:
I meant that `damnedwords.log`.txt is a fake that will work.

Nadahar replied:
@Sami32 Yes and I meant that both ```damnedwords.txt``` and ```damnedwords.log.txt``` will work, and neither is "fake".

Nadahar replied:
@Sami32 That's the point. For a 7-zip file it's fake, for a text file (which is the subject here) it's not.

Nadahar replied:
@Sami32 I've fixed the error in the text, thanks 👍

Nadahar commented:
@Sami32 I've added https://github.com/Sami32/UniversalMediaServer/commit/bb4cb85594423eb42ba62e8ac4bb553106d9725e, but there's a problem. The tests now fail because "MP4" is no longer a valid video codec. That is correct in a sense, but it should probably be made so that ```MP4``` was accepted as either ```SP``` or ```ASP```. If not, a lot of renderer configurations needs to be updated.

Sami32 commented:
Correct, i remember having fixed it later, perhaps 1 or 2 weeks ago, and have modified some renderers configuration as well.

Nadahar commented:
@Sami32 What was the fix? I haven't looked into the details, but I'd think that any renderer that can play ASP would play SP as well.

Sami32 commented:
https://github.com/UniversalMediaServer/UniversalMediaServer/commit/bb4cb85594423eb42ba62e8ac4bb553106d9725e

Not at all, just look at you PS3 as exaple. Many are in that case.

Nadahar commented:
@Sami32 I can't see any fix in the commit you linked. Can you link directly to the fix?

Sami32 commented:
I cannot remember what test i did fix, only that i did it for MP4 and cannot remember where i've rebased it...
It complile and play fine for me, so that mean that i did it somewhere...

Nadahar commented:
@Sami32 That doesn't help me very much. It's not that I can't "fix" the test, I want to know how to address the problem itself.

Sami32 commented:
Sorry i cannot remember. Maybe later. I'll look at it.

Sami32 commented:
If you could post the Maven error, it should help me remember i guess

Nadahar commented:
@Sami32 The error isn't what's important. The test failure is because the PS3 is no longer compatible with "MP4" video codec. My question is how do we solve it. Either ```MP4``` must be made an alias for one of the new codecs, or all the renderer configurations must be changed as far as I can understand.

Sami32 commented:
Sorry i wasn't here.

Why didn't you said that at first !
That why i din't find trace into the commit, i just modified the PS3 configuration as following:
`Supported = f:avi|divx v:mpeg4sp|divx|mjpeg|xvid a:mp3|lpcm|mpa|ac3 m:video/x-divx gmc:0`

Sami32 commented:
I just changed the renderers that i knew need such support. Mater it could be done automatically from informations taken from SinkProtocolInfo.
For now, i think that modifying the renderers' configuration that need it should be fine, as PS3 is the only one having a test for it, and that SP and ASP profile are mainly met in AVI files that are not very popular now.
That said, i'm open to any advise ;)

Nadahar commented:
@Sami32 All the renderer configurations with ```v:MP4``` in them will fail now. We need to edit them, but the question is what to put there. The affected configurations are these:
```
Apple-iDevice-VLC32bit.conf
Apple-TV-VLC.conf
Google-Android-BubbleUPnP-MXPlayer.conf
Google-Android-Chromecast.conf
Google-ChromecastUltra.conf
Hisense-K680.conf
Kodi.conf
LG-BP.conf
LG-BP550.conf
LG-EG910V.conf
LG-LB.conf
LG-LM620.conf
LG-LM660.conf
LG-LS5700.conf
LG-ST600.conf
LG-UB820V.conf
LG-WebOS.conf
Microsoft-WindowsMediaPlayer.conf
Microsoft-Xbox360.conf
Microsoft-XboxOne.conf
Panasonic-SCBTT.conf
Panasonic-Viera.conf
Panasonic-VieraAS600E.conf
Panasonic-VieraCX680.conf
Panasonic-VieraCX700.conf
Panasonic-VieraE6.conf
Panasonic-VieraGT50.conf
Panasonic-VieraST60.conf
Panasonic-VieraTXL32V10E.conf
Panasonic-VieraVT60.conf
Philips-PFL.conf
Pioneer-BDP.conf
Roku-Roku3-3.conf
Roku-Roku3-5.conf
Samsung-9series.conf
Samsung-BDC6800.conf
Samsung-BDH6500.conf
Samsung-C6600.conf
Samsung-D6400.conf
Samsung-EH5300.conf
Samsung-EH6070.conf
Samsung-ES6100.conf
Samsung-ES8000.conf
Samsung-ES8005.conf
Samsung-F5100.conf
Samsung-F5505.conf
Samsung-F5900.conf
Samsung-GalaxyS5.conf
Samsung-H4500.conf
Samsung-H6203.conf
Samsung-H6400.conf
Samsung-HTE3.conf
Samsung-HTF4.conf
Samsung-HU7000.conf
Samsung-HU9000.conf
Samsung-J6200.conf
Samsung-PL51E490.conf
Samsung-WiseLink.conf
Sharp-Aquos.conf
Showtime3.conf
Showtime4.conf
Sony-Bluray.conf
Sony-Bluray2013.conf
Sony-BraviaEX.conf
Sony-BraviaEX725.conf
Sony-BraviaHX.conf
Sony-BraviaHX75.conf
Sony-BraviaNX70x.conf
Sony-BraviaNX800.conf
Sony-BraviaW.conf
Sony-BraviaX.conf
Sony-BraviaXBR.conf
Sony-BraviaXD.conf
Sony-PlayStation3.conf
Sony-PlayStation4.conf
Sony-PlayStationVita.conf
Sony-SMPN100.conf
Sony-STR5800ES.conf
Sony-Xperia.conf
Sony-XperiaZ3.conf
Technisat-S1Plus.conf
Telefunken-TV.conf
Vizio-SmartTV.conf
```

Sami32 commented:
I saw it very simply, as all these renderers will disappear one day:
Just replace all the `mp4` by `mpeg4sp|mpeg4asp` in all the AVI and MOV supported lines, except for PS3 (SP only) ans PS4 (ASP only), as they can be fine tuned later if user ask for.
And in MKV and MP4 supported lines, add `mpeg4sp|mpeg4asp`

I only did it for few ones, as i only used it for testing for now.

Nadahar commented:
@Sami32 Yes, that's a possibility although it's a lot of job. Another possibility is to make a "hack" that means that ```mp4``` means ```mpeg4sp|mpeg4asp``` and just change the PS3 and PS4.

Sami32 commented:
I could do it, if you are not in a hurry ?

Sami32 commented:
The problem, is that i'm afraid that this hack will be forgotten, and that when a proper DLNA implementation will be done, it'll stay.

Nadahar commented:
@Sami32 If you're up for doing it, be my guest. You can just push a new commit here with the changes. 

When the proper implementation is made, none of the existing system will be used, so I'm not worried about that part. The current system is flawed on so many levels that there's no point in keeping any of the code as I see it. When implementing DLNA; each limitation of that profile must be verified - and I expect there to be at least as many things to check as for images. That means that a lot more details must be verified than what the current system can even "dream of" handling.

Sami32 commented:
Ok.
I know, but i try to add them little by little, as the bitdepth, colordepth, framerate...some haven't been outside of my build though.

Nadahar commented:
@Sami32 I'll try to do an automated replace with regex, so hold on before you start editing.

Sami32 commented:
I've already started some of them

Nadahar commented:
@Sami32 I put my commit in before yours so we can see the difference. It seems that you haven't removed the ```mp4``` setting. Why is that?

Sami32 commented:
Because others MP4 codecs exist, like Sony/Panasonic XAVC.
Yes, i know that i'm the only one having added the support for it ;P

Nadahar commented:
@Sami32 Ok, now I'm really confused.

Nadahar commented:
@Sami32 Another point, I just added one of the commits on "my list": https://github.com/UniversalMediaServer/UniversalMediaServer/commit/94df78fa65f461cc6a2ffa32a69ae7c1112f87bb

The problem is that when I've removed all that is already in this branch pluss the FFmpegRemux (which is already done in EnginesRegistration), I'm left with very little. Can you take a look at the commit and see if it's actually useful?

I did some refactoring in the thumbnail generation code, because it was so messy. In your commit you had done a lot of stuff with the thumbnail padding. In this branch all that code was removed, so I let it stay that way. What is the point of that? Padding for the renderer will be handled by ImagesUtil later - I don't understand why it should be done here. It's also wrong, because the generated thumbnail is cached and could be used for any renderer, regardless of their "padding" setting.

Is the problem that the thumbnail is stretched? I don't get it.

Sami32 commented:
Yes, but you never confirmed that you have reproduced it with the sample i sended you, so i stoppped to work on it. I guess that it was set when i did some reordering and rebasing, so you can safely remove it, as i've trace of it in 2 other commits.
I remember that somewhere you've also removed the thumbnail padding, so as i never tested if it was still streched with your PR, you can remove it.

Nadahar commented:
@Sami32 There were some other change there though, you replaced the "-an" with "". I refactored that part so that the thumbnail parameters aren't first added and then cleared later, which means there's no need to replace them with either "-an" or "". Now the thumbnail parameters are only added it the thumbnail should actually be generated.

Nadahar commented:
@Sami32 About the thumbnail padding and aspect ratio, the only thing FFmpeg should do is to produce a thumbnail with a correct aspect ratio. If that is done, other code will handle resizing and padding as needed.

Nadahar commented:
@Sami32 What do we do with the renderers? Do you have any idea which ones that support XAVC?

Sami32 commented:
As i said Panasonic and Sony should, but others codec as MPEG-4 v1 v2 or v3 are also more often met, so that doesn't limit to XAVC version.
